packages/terminal-extension/src/index.ts

```diff
@@ -171,7 +177,63 @@ function activate(
     });
   });
 
-  addCommands(app, tracker, settingRegistry, translator, options);
+  function stdoutTransformer(output: string): string {
+    let outputJSON;
+    if (output.indexOf(MAGIC_OPEN_STRING) === 0) {
+      let outputJSON = JSON.parse(output.replace(MAGIC_OPEN_STRING, ''));
```

Comment:
Are we guaranteed that these magic strings will always be at the start of a message? I'm thinking of tricky buffering that may split up the stdout text (and hence stdout messages) in unpredictable ways.

Reply:
Guaranteed until we are not :-) I think using control codes on both boundaries of the JSON will be much safer.